Source/WebCore/platform/ScrollableArea.cpp

```diff
@@ -207,9 +216,9 @@ void ScrollableArea::scrollToOffsetWithoutAnimation(ScrollbarOrientation orienta
 {
     auto currentPosition = scrollAnimator().currentPosition();
     if (orientation == ScrollbarOrientation::Horizontal)
-        scrollAnimator().scrollToPositionWithoutAnimation(FloatPoint(offset, currentPosition.y()));
+        scrollAnimator().scrollToPositionWithoutAnimation(FloatPoint(offset - scrollOrigin().x(), currentPosition.y()));
```

Comment:
Please rewrite to use the helper `scrollPositionFromOffset()`.

Reply:
I'm not sure the addition of `scrollToPositionWithoutAnimation(, float)` is justified, especially since it just does the `if (orientation == ScrollbarOrientation::Horizontal)` check again.